src/Resources/ResourceCollection.php

```diff
@@ -3,80 +3,61 @@
 namespace DeliciousBrains\SpinupWp\Resources;
 
 use Countable;
-use DeliciousBrains\SpinupWp\Endpoints\Endpoint;
 use DeliciousBrains\SpinupWp\SpinupWp;
 use Generator;
 use IteratorAggregate;
 
 class ResourceCollection implements Countable, IteratorAggregate
 {
-    protected array $payload;
-
     protected string $class;
 
-    protected Endpoint $endpoint;
-
     public SpinupWp $spinupwp;
 
-    protected int $page;
-
-    protected array $data;
-
-    public function __construct(array $payload, string $class, Endpoint $endpoint, SpinupWp $spinupwp, int $page = 1)
-    {
-        $this->payload  = $payload;
-        $this->class    = $class;
-        $this->endpoint = $endpoint;
-        $this->spinupwp = $spinupwp;
-        $this->page     = $page;
+    protected ?Paginator $paginator;
 
-        $this->mapResourceClass();
-    }
+    protected array $resources = [];
 
-    protected function mapResourceClass(): void
+    public function __construct(array $resources, string $class, SpinupWp $spinupwp, ?Paginator $paginator = null)
     {
-        $this->data = array_map(function ($data) {
-            return new $this->class(['data' => $data], $this->spinupwp);
-        }, $this->payload['data']);
-    }
+        $this->class     = $class;
+        $this->spinupwp  = $spinupwp;
+        $this->paginator = $paginator;
 
-    protected function hasNext(): bool
-    {
-        return !empty($this->payload['pagination']['next']);
+        $this->resources = $this->mapResourceClass($resources);
     }
 
-    protected function hasPrevious(): bool
+    protected function mapResourceClass(array $resources): array
```

Comment:
This function with have unexpected results if `$this->class` does not extend `DeliciousBrains\SpinupWp\Resources\Resource`.

This is probably not incredibly important in the context of this project but I can't help but think that these should be more tightly coupled. Though I can't think of an obvious solution.

Reply:
If we were expecting `ResourceCollection` to be used externally by users, I would agree. However, I don't see any reason why someone would expect this to work outside the context of passing list API data.